clang/lib/AST/ODRHash.cpp

```diff
@@ -14,6 +14,11 @@
 
 #include "clang/AST/ODRHash.h"
 
+#include "clang/AST/APValue.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/CharUnits.h"
+#include "clang/AST/Decl.h"
+#include "clang/AST/DeclCXX.h"
```

Comment:
Do we actually need these headers?

Reply:
Types from all of them are used inside `AddStructuralValue` function (`ASTContext` and `CharUnits` implicitly, as intermediate return types inside `D->getASTContext().getMemberPointerPathAdjustment(Value).getQuantity())` expression.

Reply:
Removed.